### ISSUE TITLE
[Bug]: Advanced customization issues

### ISSUE THREAD

**Issue by artyorsh** [braze-inc/braze-react-native-sdk#216]
### Which Platforms?

iOS

### Which React Native Version?

0.71.8

### Which @braze/react-native-sdk SDK version?

5.0.0

### Repro Rate

100%

### BrazeInAppMessageUIDelegate conflict

When following [IAM/Advanced customization](https://www.braze.com/docs/developer_guide/platform_integration_guides/react_native/inapp_messages/#advanced-customization)  guide, the app is still not able to receive IAM. The root of the cause is [this line](https://github.com/braze-inc/braze-react-native-sdk/blob/master/iOS/BrazeReactBridge/BrazeReactBridge/BrazeReactBridge.m#L85), which is called within `Braze.addListener`. It overrides the inAppMessagePresenter assigned from native runtime (Swift), making native delegate useless. Commenting the line made it work for our app, but I think we need a nullability check there.

### TS issue

Defining a listener like this, produces a TS error when following Advanced customization guide. 
```ts
Braze.addListener(Braze.Events.IN_APP_MESSAGE_RECEIVED, (event) => {
  const msg = JSON.parse(event.inAppMessage);
});
```

The root of the cause, I believe, is in [this line](https://github.com/braze-inc/braze-react-native-sdk/blob/master/src/index.d.ts#L797), which assumes standard setup.

### Expected Behavior

Advanced customization is compatible to standard setup.

### Actual Incorrect Behavior

Braze.addListener overrides IAM delegate configured from native side

**Comment by hokstuff** [comment]
Hi @artyorsh,

We just released [React Native SDK 5.2.0](https://github.com/braze-inc/braze-react-native-sdk/releases/tag/5.2.0) which contains a fix for this issue by introducing a nullability check. Let us know if this solves your issue.

Thanks!

**Comment by artyorsh** [comment]
Thanks for the update @hokstuff. It works for us now 👍
